runtime/compiler/z/codegen/J9S390CHelperLinkage.cpp

```diff
@@ -364,6 +361,17 @@ TR::Register * TR::S390CHelperLinkage::buildDirectDispatch(TR::Node * callNode,
       }
    TR::SymbolReference * callSymRef = callNode->getSymbolReference();
    void * destAddr = callNode->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethodAddress();
+
+#if defined(J9ZOS390)
```

Comment:
@NigelYiboYu I think my previous comment was for this section. For now we need this part for calling out helpers written in C that means we need to load environment for instanceOf as well. Is that covered? Looking through your other changes. 
 I am going through your changes now.

Reply:
@fjeremic I am thinking of moving some of the context switching code from helper Glue code to in body and see the effect. I expect a small effect on the method footprint. If the effect is not much then we can move this inlined. I believe x86 already do this. We can make our helper linkage more generic this way.

Reply:
`InstanceOf` is not covered in this PR. This whole pull request is for JNI and IFA switching. We can come back and change `instanceOf` helper to load environment.

Reply:
Actually good point @r30shah. Thinking about this again we have duplicate information here. The helper index is already stored in `callNode->getSymbolReference()->getReferenceNumber()` so we shouldn't need to pass it as an argument to the CHelper linkage `buildDirectDispatch`.

This information is already available through the `callNode`. @NigelYiboYu are we able to get rid of this extra parameter? Once this is done we can remove the below check for `if (helperIndex != TR_S390numRuntimeHelpers)` since a helper index must always be available. This will fix the `instanceOf` helper as well.

Reply:
The extra helper index is removed.

Reply:
Update:

Environment pointer is now only loaded if the C helper is fast-path. Non-fastPath helpers jump to glue code, which does its own environment handling.